test/envConfig.test.ts

```diff
@@ -49,3 +58,109 @@ describe('createEnvConfigContent', () => {
     expect(result).toMatchSnapshot();
   });
 });
+
+describe('envConfig', () => {
+  const FIXTURES_PATH = path.join(__dirname, 'fixtures');
+
+  describe('devServer', () => {
+    let server: ViteDevServer;
+    const env = process.env;
+
+    beforeEach(() => {
+      jest.resetModules();
+      process.env = { ...env };
+    });
+
+    afterEach(async () => {
+      process.env = env;
+      await server.close();
+    });
+
+    it('serves /index.html that includes envConfig.js script', async () => {
+      expect.assertions(2);
+
+      // given
+      server = await createServer({
+        root: FIXTURES_PATH,
+        plugins: [envConfig()],
+        server: {
+          port: 3000,
+        },
+      });
+      await server.listen();
+
+      // when
+      const response = await request('http://localhost:3000').get('/index.html');
+
+      // then
+      expect(response.status).toBe(200);
+      expect(response.text).toContain('<script src="/env-config.js"></script>');
+    });
+
+    it('serves /env-config.js', async () => {
+      expect.assertions(2);
+
+      // given
+      const variables = ['BACKEND_URL'];
+      process.env.BACKEND_URL = 'http://localhost:4000';
```

Comment:
```suggestion
      const myEnv = { BACKEND_URL: 'http://localhost:4000' };
      const variables = Object.keys(myEnv);
      process.env = { ...process.env, ...myEnv };
```

Reply:
I will not do this for now as it makes the type of `process.env.BACKEND_URL` `string | undefined` which would complicate the test further.

Furthermore, I think this is generally less readable than just duplicating the key :thinking: